lib/dialects/abstract/connection-manager.js

```diff
@@ -273,6 +274,7 @@ class ConnectionManager {
     return promise.then(() => {
       return this.pool.acquire(options.priority, options.type, options.useMaster)
         .then(mayBeConnection => this._determineConnection(mayBeConnection))
+        .catch(err => err.name === 'TimeoutError', err => { throw new errors.ConnectionAcquireTimeoutError(err); })
```

Comment:
Short version `catch('TimeoutError', err => { ... })`

Add this check to https://github.com/sequelize/sequelize/blob/342e18fbaddb74be39d6ea1dd117cb291f1c65e0/lib/dialects/abstract/connection-manager.js#L154 and https://github.com/sequelize/sequelize/blob/342e18fbaddb74be39d6ea1dd117cb291f1c65e0/lib/dialects/abstract/connection-manager.js#L157

Reply:
Don't know whether you've seen my question: it seems that acquire() method of replication pool proxy object is never called directly except from getConnection() method, where all necessary checks are made, so wouldn't be catch() here and moreover check for mayBeConnection redundant?

Reply:
You are right, for both replication and general case, `this.pool.acquire` in `getConnection` represent all calls. Please go ahead and remove `mayBeConnection` check from replication pool proxy's `acquire` case.

Apart from that just simplify `catch('TimeoutError', err => { ... })` and this should be good to go

Reply:
Removed checks.
Considering catch() simplification: string didn't worked. It coul be a regex, but then it tries to match with whole stack, therefore fails test for errors in mssql/connection-manager. Found in bluebird docs the way to make it a bit shorter with object predicate.